### ISSUE TITLE
 Request to increase maximum supported Python version to 3.12

### ISSUE THREAD

**Issue by myuanz** [ZEISS/pylibczirw#47]
The current `setup.py` limits Python support to 3.11, but I've successfully compiled and tested the project with Python 3.12.

Could you please consider updating the `setup.py` file to include support for Python 3.12?

**Comment by hotzenklotz** [comment]
@sebi06 Is there any news on that? Python 3.13 was just released on this lib is not even compatible with Python 3.12 yet.

**Comment by CraigZeiss** [comment]
some more info in this issue #54

**Comment by pgarrison** [comment]
I'm adding support to bioio for reading CZIs using pylibczirw instead of aicspylibczi, but we'd like to support python >= 3.9 in bioio.

I'm planning to submit a PR to remove the python version upper bound from pylibczirw. Do any of the pylibczirw maintainers know of concrete barriers to using pylibczirw on Python 3.12 or 3.13 or greater?

**Comment by sebi06** [comment]
Hi @pgarrison,

I am not really aware of any strong "barrier" etc. on our side. AFAIK it is mainly the fact that for our use cases we mainly use python 3.9 and 3.10.  In the end it was mainly resources and priorities on our side not to tackle this yet. But a PR is of course welcome

**Comment by pgarrison** [comment]
I have successfully tested pylibczirw against python 3.12 and 3.13 as follows:

* I ran the pylibczirw tests by triggering the test github action from my fork (it seems the tests are only configured to run on Windows)
* I ran the [bioio-czi](https://github.com/bioio-devs/bioio-czi) tests against pylibczirw using some not-yet-merged modifications to the bioio-czi reader.

However, I cannot make a PR yet because I am waiting on management approval to sign the CLA. In this case, the changes themselves are very small, so I encourage someone else to make a PR removing the upper bound on the supported python version.

**Comment by sebi06** [comment]
Hi @pgarrison did you just run all the python tests or did you also try to build a new wheel etc. And is there a preview on your proposed changes?

**Comment by pgarrison** [comment]
Yes, I did build wheels for 3.12 and 3.13. See the changes & github action run here: https://github.com/pgarrison/pylibczirw/pull/2

**Comment by CraigZeiss** [comment]
@pgarrison @sebi06 
i have a similar pr #64, as i didnt see the other one in the fork. I can either continue mine, or @pgarrison can merge the changes to limit to 3.13?

**Comment by pgarrison** [comment]
@CraigZeiss Please continue yours as I have not yet gotten approval to sign the CLA.

**Comment by pgarrison** [comment]
I think this has been resolved by #64, published in v5.0.0
